From 26d9c18b835b9c9a08abacec2310bfd4ba888ff2 Mon Sep 17 00:00:00 2001
Message-Id: <26d9c18b835b9c9a08abacec2310bfd4ba888ff2.1426574757.git.chang-joon.lee@intel.com>
In-Reply-To: <eba22108096acc90a0516b200c616db90ae0cb5f.1426574757.git.chang-joon.lee@intel.com>
References: <eba22108096acc90a0516b200c616db90ae0cb5f.1426574757.git.chang-joon.lee@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Thu, 12 Mar 2015 10:08:02 +0800
Subject: [PATCH 3/8] Revert "REVERTME [VPG]: drm/i915: wa for mipi cmd mode
 port reg cannot be read in CHV"

This reverts following, since the mipi port reading bug is in C0 which
will be out on market, we need the workaround to upstream.

	commit 243d34ffa6e9443d165cf3425f2014c5d7f8fcb1
	Author: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
	Date:   Mon Dec 15 18:43:51 2014 +0530

	REVERTME [VPG]: drm/i915: wa for mipi cmd mode port reg cannot be read in CH

Issue: IMINAN-22667
Change-Id: Id5962873bffca9f872e18f1cc4b8beec24a7ecdf
Signed-off-by: Hong Liu <hong.liu@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |   35 ++++++++---------------------------
 drivers/gpu/drm/i915/intel_dsi.h |    3 ---
 2 files changed, 8 insertions(+), 30 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 33a5e8c..319abe8 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -220,15 +220,9 @@ static void intel_dsi_port_enable(struct intel_encoder *encoder)
 
 		I915_WRITE(MIPI_PORT_CTRL(0), temp);
 
-		if (pipe && IS_CHERRYVIEW(dev_priv->dev) && STEP_TO(STEP_B3)) {
-			intel_dsi->port_ctrl_reg_val |= port_control;
-			I915_WRITE(MIPI_PORT_CTRL(1),
-						intel_dsi->port_ctrl_reg_val);
-		} else {
-			temp = I915_READ(MIPI_PORT_CTRL(1));
-			temp = temp | port_control;
-			I915_WRITE(MIPI_PORT_CTRL(1), temp);
-		}
+		temp = I915_READ(MIPI_PORT_CTRL(1));
+		temp = temp | port_control;
+		I915_WRITE(MIPI_PORT_CTRL(1), temp);
 
 		if ((intel_dsi->dual_link & MIPI_DUAL_LINK_FRONT_BACK) &&
 			((IS_CHERRYVIEW(dev_priv->dev) && STEP_FROM(STEP_B0)) ||
@@ -245,17 +239,10 @@ static void intel_dsi_port_enable(struct intel_encoder *encoder)
 	/* Enable the ports */
 
 	do {
-		if (pipe && IS_CHERRYVIEW(dev_priv->dev) && STEP_TO(STEP_B3)) {
-			intel_dsi->port_ctrl_reg_val |= is_cmd_mode(intel_dsi) ?
-					TEARING_EFFECT_GPIO : DPI_ENABLE;
-			I915_WRITE(MIPI_PORT_CTRL(pipe),
-						intel_dsi->port_ctrl_reg_val);
-		} else {
-			temp = I915_READ(MIPI_PORT_CTRL(pipe));
-			temp |= is_cmd_mode(intel_dsi) ?
+		temp = I915_READ(MIPI_PORT_CTRL(pipe));
+		temp |= is_cmd_mode(intel_dsi) ?
 					TEARING_EFFECT_GPIO : DPI_ENABLE;
-			I915_WRITE(MIPI_PORT_CTRL(pipe), temp);
-		}
+		I915_WRITE(MIPI_PORT_CTRL(pipe), temp);
 
 		pipe = PIPE_B;
 	} while (--count > 0);
@@ -462,14 +449,8 @@ static void intel_dsi_port_disable(struct intel_encoder *encoder)
 			(I915_READ(MIPI_PORT_CTRL(pipe)) & DPI_ENABLE))
 		dev_priv->video_disabled = true;
 
-	if (pipe && IS_CHERRYVIEW(dev_priv->dev) && STEP_TO(STEP_B3)) {
-		/* cht hw issue that MIPI port C reg cannot be read */
-		I915_WRITE(MIPI_PORT_CTRL(pipe), 0);
-		intel_dsi->port_ctrl_reg_val = 0;
-	} else {
-		I915_WRITE(MIPI_PORT_CTRL(pipe), 0);
-		POSTING_READ(MIPI_PORT_CTRL(pipe));
-	}
+	I915_WRITE(MIPI_PORT_CTRL(pipe), 0);
+	POSTING_READ(MIPI_PORT_CTRL(pipe));
 
 	if (intel_dsi->dual_link) {
 		I915_WRITE(MIPI_PORT_CTRL(pipe ? 0 : 1), 0);
diff --git a/drivers/gpu/drm/i915/intel_dsi.h b/drivers/gpu/drm/i915/intel_dsi.h
index d437c3d..46be640 100644
--- a/drivers/gpu/drm/i915/intel_dsi.h
+++ b/drivers/gpu/drm/i915/intel_dsi.h
@@ -508,9 +508,6 @@ struct intel_dsi {
 	u8 dual_link;
 	u8 pixel_overlap;
 
-	/* cht hw issue that MIPI port C reg cannot be read */
-	u32 port_ctrl_reg_val;
-
 	struct drm_i915_gem_object *gem_obj;
 	void *cmd_buff;
 	dma_addr_t cmd_buff_phy_addr;
-- 
1.7.9.5

